src/components/Gif.js

```diff
@@ -0,0 +1,101 @@
+import React, {
+    Component
+} from "react";
+import {
+    addFav,
+    withdrawFav
+} from "../Containers/FavPage/action"
+import {
+    connect
+} from 'react-redux'
+
+import {Wraper, ImgStyled} from "./Gif.style";
+
+class Gifs extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            buttonText: "Fav",
+            fav: false
+        }
+        this.onClick = this.onClick.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.gif.length > 0) {
+            this.props.gif.map((image) => {
+                if (image.id === this.props.id) {
+                    this.setState({
+                        fav: true,
+                        buttonText: "NotFav"
+                    });
+                }
+            });
+        }
+    }
+
+    onClick() {
+        if (this.props.gif.length > 0) {
+            this.props.gif.map((image) => {
+                if (image.id === this.props.id) {
+                    this.setState({
+                        fav: true
+                    });
+                }
+            });
+            if(this.state.fav){
+                this.props.withdrawFav(this.props.id);
+                this.setState({
+                    fav: false,
+                    buttonText:"Fav"
+                });
+            } else{ this.props.addFav({
+                id: this.props.id,
+                url: this.props.url
+            });
+            this.setState({
+                fav: true,
+                buttonText:"NotFav"
+            });
+            }
+        } else {
+            this.props.addFav({
+                id: this.props.id,
+                url: this.props.url
+            });
+            this.setState({
+                fav: true,
+                buttonText:"NotFav"
+            });
+        }
+
+    }
+
+    render() {
+        return ( 
+            < React.Fragment >
+            <Wraper onClick = {this.onClick}>
+                <ImgStyled src={require("./"+this.state.buttonText+".png")}   />
```

Comment:
Why you have 2 img?

Reply:
One is a heart button and the other one is the gif

Reply:
oh, you are right. this.state.buttonText isn't a good place to save the heart url. You can just create a const in the top of the file ;) or..... you can create an Heart component or an localImg component. Let's create the legos :D